resources/design-document.md

```diff
@@ -0,0 +1,244 @@
+
+Brecghan's Training Matrix(BTM) will also provide a web interface for users to manage
+the matrix. A main page providing connections to view trainings, employees, or test results. From those pages Users
+will be able to create new items and link off to the other viewing pages.
+
+# 5. API
+
+## 6.1. Public Models
+
+```
+// EmployeeModel
+
+String employeeId;
+String employeeName;
+Boolean isActive;
+Enum team;
+Date startDate;
+Set<String> trainingsTaken;
+Set<String> testsTaken;
+Enum trainingStatus;
+```
+
+```
+// TrainingModel
+
+String trainingId;
+String trainingName;
+ZoneDateTime trainingDate;
+Boolean isActive;
+Integer monthsTilExpire;
+Enum expirationStatus;
+Set<String> testsForTraining;
+Set<String> employeesTrained;
+String trainingSeries;
+
+```
+
+```
+// TestModel
+
+String trainingId;
+String employeeId;
+Integer scoreToPass;
+Integer latestScore;
+Boolean hasPassed;
+Integer timesTaken;
+```
+
+```
+// TrainingSeriesModel
+
+String trainingSeriesName;
+```
+
+## 6.2. Endpoints
+
+### Employees
+
+* Accepts `POST` requests to `/employee`
+* Accepts an employeeId, employeeName, start Date, and team; returns the corresponding EmployeeModel including a true boolean parameter isActive, a training status of Up-To-Date, and 2 blank sets of trainingsTaken and testsTaken.
+
+* Accepts `GET` requests to `/employee`
+* Returns a list of all employees.
+
+* Accepts `GET` requests to `/employee?isActive={isActive}`
+* Returns a list of employees. If specified through the boolean query parameter, returns only Active employees.
+
+* Accepts `GET` requests to `/employee?team={team}`
+* Accepts a team and returns the employees within the corresponding team. 
+ 
+* Accepts `GET` requests to `/employee/{employeeId}`
+* Accepts an employeeId and returns the corresponding EmployeeModel.
+
+* Accepts `PUT` requests to `/employee/{employeeId}`
+* Accepts data to update an employee including an updated employeeName, team, and isActive boolean.
+
+* Accepts `GET` requests to `/employee/{employeeId}/tests`
+* Accepts an employeeId returns the corresponding list of tests for that employee.
+
+### Training Series
+
+* Accepts `POST` requests to `/trainingSeries`
+* Accepts a trainingSeriesName and returns a list of Training Series, containing the newly added Series.
+
+* Accepts `GET` requests to `/trainingSeries`
+* Returns a list of Training Series.
+
+* Accepts `GET` requests to `/trainingSeries/trainings`
+* Returns a list of all trainings for a specific Training Series.
+
+### Trainings
+
+* Accepts `POST` requests to `/training`
+* Accepts a trainingName, trainingDate, trainingSeries(can be null), and monthsTilExpire and returns the corresponding TrainingModel including a unique trainingId assigned by the Training Service.
+
+* Accepts `GET` requests to `/training`
+* Returns a list of all Trainings.
+
+* Accepts `GET` requests to `/training?isActive={isActive}`
```

Comment:
Same idea for this and the next one as the note above on employee - include the query parameters in the top level `/training` endpoint.